astropy/table/serialize.py

```diff
@@ -0,0 +1,165 @@
+                # object constructors don't pay attention to the .info attributes
+                # in the input data args (e.g. Quantity(col) does look at the unit
+                # but not info.description).
+                if len(data_attrs) > 1:
+                    name = name + '.' + data_attr
+                    col_info = {}
+                else:
+                    col_info = info
+                new_cols.append(Column(data, name=name, **col_info))  # TODO masking, MaskedColumn
+
+                obj_attrs[data_attr] = SerializedColumn({'name': name})
+
+            # Store the fully qualified class name
+            obj_attrs['__class__'] = col.__module__ + '.' + col.__class__.__name__
+
+            if info:
+                # Strip out any attributes defined by the parent
+                for attr in col.info.attrs_from_parent:
+                    if attr in info:
+                        del info[attr]
+                obj_attrs['__info__'] = info
+
+            mixin_cols[col.info.name] = obj_attrs
+
+    meta = deepcopy(tbl.meta)
+    meta['__mixin_columns__'] = mixin_cols
+    out = Table(new_cols, meta=meta, copy=False)
+
+    return out
+
+
+def _construct_mixin_from_obj_attrs(obj_attrs):
+    cls_full_name = obj_attrs.pop('__class__')
+
+    # If this is a supported class then import the class and run
+    # the _construct_from_col method.  Prevent accidentally running
+    # untrusted code by only importing known astropy classes.
+    if cls_full_name not in __construct_mixin_classes:
+        raise ValueError('unsupported class for construct {}'.format(cls_full_name))
+
+    mod_name, cls_name = re.match(r'(.+)\.(\w+)', cls_full_name).groups()
+    module = import_module(mod_name)
+    cls = getattr(module, cls_name)
+    return cls.info._construct_from_dict(obj_attrs)
+
+
+def _construct_mixins_from_columns(tbl):
+    if '__mixin_columns__' not in tbl.meta:
+        return tbl
+
+    out = tbl.copy(copy_data=False)
+
+    mixin_cols = out.meta.pop('__mixin_columns__')
+
+    for new_name, obj_attrs in mixin_cols.items():
+        data_attrs_map = {}
+        for name, val in obj_attrs.items():
+            if isinstance(val, SerializedColumn):
+                data_attrs_map[val['name']] = name
+
+        for name in data_attrs_map.values():
+            del obj_attrs[name]
+
+        # Get the index where to add new column
+        idx = min(out.colnames.index(name) for name in data_attrs_map)
+
+        # Name is the column name in the table (e.g. "coord.ra") and
+        # data_attr is the object attribute name  (e.g. "ra").  A different
+        # example would be a formatted time object that would have (e.g.)
+        # "time_col" and "value", respectively.
+        for name, data_attr in data_attrs_map.items():
+            obj_attrs[data_attr] = out[name]
+            del out[name]
+
+        info = obj_attrs.pop('__info__', {})
```

Comment:
So, here one would again want to distinguish between the single-column and the multi-column case, getting `info` from the serialized column in the single-column case (but excluding empty things).